### PR TITLE
feat(bigquery): add DataGovernanceType to routines

### PR DESCRIPTION
**Pull request by shollyman**
PR adds the DataGovernanceType field to routine metadata representations.  Testing-wise, this PR also makes test inputs more explicit.

internal issue: 286117986

### CODE REVIEW

File: bigquery/routine_test.go [hunk 1]
```diff
@@ -105,55 +113,73 @@ func TestRoutineTypeConversions(t *testing.T) {
...
 				NullFields:      []string{"Arguments", "ImportedLibraries"},
-			}},
-		{"empty", "ToRoutineArgument",
-			&bq.Argument{},
-			&RoutineArgument{}},
-		{"basic", "ToRoutineArgument",
-			&bq.Argument{
+			},
+		},
+		{name: "empty",
```
Comment (Linchin) [inline bigquery/routine_test.go:150]: nit: line break here?

**Comment by Linchin** [comment]
I wonder if it makes sense to add this field to integration tests too?

File: bigquery/routine_test.go [hunk 2]
```diff
@@ -72,9 +72,16 @@ func TestRoutineTypeConversions(t *testing.T) {
 		in         interface{}
 		want       interface{}
 	}{
-		{"empty", "ToRoutineMetadata", &bq.Routine{}, &RoutineMetadata{}},
-		{"basic", "ToRoutineMetadata",
-			&bq.Routine{
+		{
```
Comment (Linchin) [inline bigquery/routine_test.go:75]: It looks like the method `RoutineMetadata.toBQ()` isn't tested. I wonder if we need to add unit tests for it.

File: bigquery/routine_test.go (hunk 1 above)
Comment (shollyman) [reply bigquery/routine_test.go]: done.

File: bigquery/routine_test.go (hunk 2 above)
Comment (shollyman) [reply bigquery/routine_test.go:81]: added

**Comment by Linchin** [comment]
Also, I don't think I'm able to approve this PR. I'm not in the go reviewer group.

**Comment by shollyman** [comment]
> I wonder if it makes sense to add this field to integration tests too?

done.

**Review by Linchin — APPROVED** [review]
Thank you!

**Pull request merged**